lms/static/js/dashboard/legacy.js

```diff
@@ -95,29 +124,73 @@
          });
 
          $('.action-email-settings').click(function(event) {
-             var element = $(event.target);
-             $('#email_settings_course_id').val(element.data('course-id'));
-             $('#email_settings_course_number').text(element.data('course-number'));
+             $('#email_settings_course_id').val($(event.target).data('course-id'));
+             $('#email_settings_course_number').text($(event.target).data('course-number'));
              if ($(event.target).data('optout') === 'False') {
                  $('#receive_emails').prop('checked', true);
              }
              edx.dashboard.dropdown.toggleCourseActionsDropdownMenu(event);
          });
-
          $('.action-unenroll').click(function(event) {
-             var element = $(event.target);
-             var track_info = element.data('track-info');
-             var course_number = element.data('course-number');
-             var course_name = element.data('course-name');
-             var cert_name_long = element.data('cert-name-long');
-             $('#track-info').html(interpolate(track_info, {
-                 course_number: "<span id='unenroll_course_number'>" + course_number + '</span>',
-                 course_name: "<span id='unenroll_course_name'>" + course_name + '</span>',
-                 cert_name_long: "<span id='unenroll_cert_name'>" + cert_name_long + '</span>'
-             }, true));
-             $('#refund-info').html(element.data('refund-info'));
-             $('#unenroll_course_id').val(element.data('course-id'));
-             edx.dashboard.dropdown.toggleCourseActionsDropdownMenu(event);
+             var isPaidCourse = $(event.target).data('course-is-paid-course') === 'True';
+             var certNameLong = $(event.target).data('course-cert-name-long');
+             var enrollmentMode = $(event.target).data('course-enrollment-mode');
+
+             var courseNumber = $(event.target).data('course-number');
+             var courseName = $(event.target).data('course-name');
+             var courseRefundUrl = $(event.target).data('course-refund-url');
+             var showRefundOption = '';
+             var dialogMessageAttr;
+
+             var request = $.ajax({
+                 url: courseRefundUrl,
+                 method: 'GET',
+                 dataType: 'json'
+             });
+             request.success(function(data, textStatus, xhr) {
+                 if (xhr.status === 200) {
+                     showRefundOption = data.course_refundable_status;
```

Comment:
#nitpick Doesn't look like the variable `showRefundOption` is needed here. We can just pass the `data.course_refundable_status` into the `setDialogAttributes` function.